site/content/en/docs/Guides/feature-stages.md

```diff
@@ -0,0 +1,131 @@
+  If you have multiple clusters that can be upgraded independently, you may be able to relax this restriction.
+
+> Note: Please do try `Beta` features and give feedback on them! After they exit beta, it may not be practical for us
+> to make more changes.
+
+### Stable
+
+A `Stable` feature means:
+
+* The feature is enabled and the corresponding feature gate no longer exists.
+* Stable versions of features will appear in released software for many subsequent versions.
+
+## Feature Stage Indicators
+
+There are a variety of features with Agones, how can we determine what stage each feature is in?
+
+Below are indicators for each type of functionality that can be used to determine the feature stage for a given aspect
+of Agones.
+
+### Custom Resource Definitions (CRDs)
+
+This refers to Kubernetes resource for Agones, such as `GameServer`, `Fleet` and `GameServerAllocation`.
+
+#### New CRDs
+
+For new resources, the stage of the resource will be indicated by the `apiVersion` of the resource.
+
+For example: `apiVersion: "agones.dev/v1"` is a `stable` resource, `apiVersion: "agones.dev/v1beta1"` is a `beta`
+ stage resource, and `apiVersion: "agones.dev/v1alpha1"` is an `alpha` stage resource.
+
+#### New CRD attributes
+
+`alpha` and `beta` attributes will have a corresponding `alpha` or `beta` parent element in their configuration to
+delineate their feature stage.
+
+If no such parent exists, this attribute is a `stable` feature.
+
+For example, if we were to add a hypothetical `alpha` configuration option of `timeoutSeconds` to the `GameServer`
+ resource, it would be configured like so:
+ 
+```yaml
+apiVersion: "agones.dev/v1"
+kind: GameServer
+metadata:
+  generateName: "simple-udp-"
+spec:
+  alpha: # this is the alpha feature block
+    timeoutSeconds: 30
+  ports:
+  - name: default
+    portPolicy: Dynamic
+    containerPort: 7654
+  template:
+    spec:
+      containers:
+      - name: simple-udp
+        image: gcr.io/agones-images/udp-server:0.15
+``` 
+
+As resource attributes progress through their stages, there may be breaking changes, as backward conversion between
+ `alpha`, `beta` and `stable` positioning in the resource configuration may not be guaranteed.
+
+### Agones Game Server SDK
+
+Any `alpha` or `beta` Game Server SDK functionality will be a subpackage of the `sdk` package. For example
+, functionality found in a `sdk.alphav1` package should be considered at the `alpha` feature stage. 
+
+Any other SDK functionality not marked as `alpha` or `beta` can be assumed to be `stable`.
+
+### REST & gRPC APIs 
+
+REST and gRPC API will have versioned paths were appropriate to indicate their feature stage.
+
+For example, a REST API with a prefix of `v1alpha1` is an `alpha` stage feature: 
+`http://api.example.com/v1alpha1/exampleaction`.
+
+Similar to the SDK, Any `alpha` or `beta` gRPC functionality will be a subpackage of the main API package.
```

Comment:
We should perhaps mention that alpha/beta features are a super-set of the main package?

Reply:
So you are saying all stable features should be available in the alpha/beta sdk subpackage?

So I should be able to do `sdk.alphav1.Ready()`? 

If so, I'm not sure what the benefit would be?

Reply:
> So you are saying all stable features should be available in the alpha/beta sdk subpackage?
> 
I think now I understand why you wanted to run two SDK packages side by side because one is supporting a none stable API and the other supports a stable API. If this is what you are recommending, please make is clear. Otherwise, I assumed alpha API would be a super set of stable API and there is only one SDK running. So if sdk is now supporting an alpha feature SelfNominate(), I thought I create a new snapshot of the SDK and version it as sdk.v1alpha1 which is a super set of sdk[.v1] package and expose it for experimentation.

Reply:
Aaah! Yes! That is the disconnect. Yes, that is exactly the way I figured it would work, but I wasn't clear. Let me write words to that effect (it sounds like it will be similar to what we have in CRDs as well).

We can also do something like keep alpha/beta SDKs in place for a period after they move to stable, so that people have time to move over as well :+1: